src/psyclone/dynamo0p3.py

```diff
@@ -8604,6 +8548,19 @@ def __init__(self, kernel_args, arg_meta_data, arg_info, call):
         # it. Note, issue #79 is also related to this.
         KernelArgument.__init__(self, arg_meta_data, arg_info, call)
 
+        # Name of the LFRic infrastructure module that declares the argument
+        # data type
+        self._data_module = None
+        # Argument data type (if/as defined in the LFRic infrastructure)
+        self._data_type = None
+        # Argument proxy data type (if/as defined in LFRic infrastructure)
+        self._proxy_data_type = None
+        # Use 'infer_datatype' to set up kernel argument information: module
+        # name, data type, proxy data type and precision (precision is
+        # already initialised to 'None' in the parent class). This is
+        # currently supported for scalars, fields and operators.
+        _ = self.infer_datatype(proxy=False)
```

Comment:
I created a new function to do this as requested. I did not revert the code completely for `infer_datatype` as it can now access the requested information about precision, data type and module from the initialiser.

Reply:
P.S. Creating a single instance of `LFRicConstants()` in the initialisation using it within the rest of the class doesn't work. There are other examples of creating instances in class methods and not initialiser. BTW, isn't `LFRicConstants()` supposed to be a singleton so a reference will not create a new instance if it does not already exist?

Reply:
`LFRicConstants` only uses class variables. These are defined/read/computed the first time an instance is created, from then on the constructor basically returns immediately:
```
    def __init__(self):
        if LFRicConstants.HAS_BEEN_INITIALISED:
            return
        LFRicConstants.HAS_BEEN_INITIALISED = True
        ...
```
That fact that the constructor is so cheap (except the first time) is why I never bothered storing the instance, usually I would just get a new one.

I can't say why creating one instance in the constructor and keeping it does not work - I think it should. There's #1295 (using LFRicConstants and config during import time, which can fail if psyclone is not installed and PSYCLONE_CONFIG is not set), but I can't see how that could be related.